src/openms/include/OpenMS/KERNEL/MSSpectrumSoAWrapper.h

```diff
@@ -137,12 +189,58 @@ struct BaseContainer
         resize(size_);
     }
 
+    bool empty()
+    {
+        if(this->size()==0)
+            return true;
+        
+        return false;
```

Comment:
```suggestion
        return this->size() == 0;
```

or even `!this->size()`